src/games/strategy/sound/ClipPlayer.java

```diff
@@ -268,10 +272,12 @@ private void playClip(final String clipName, final PlayerID playerId) {
     if (clip != null) {
       (new Thread(() -> {
         try {
-          final AdvancedPlayer player = new AdvancedPlayer(
-              new java.net.URL(clip.toURL().toString()).openStream(),
-              FactoryRegistry.systemRegistry().createAudioDevice());
-          player.play();
+          Optional<InputStream> inputStream = UrlStreams.openStream(clip.toURL());
+          if(inputStream.isPresent()) {
+            final AudioDevice audioDevice = FactoryRegistry.systemRegistry().createAudioDevice();
+            final AdvancedPlayer audioPlayer = new AdvancedPlayer(inputStream.get(), audioDevice);
+            audioPlayer.play();
```

Comment:
Here you could simplify to `new AdvancedPlayer(inputStream.get(), audioDevice).play();`